csrank/objectranking/lambdamart.py

```diff
@@ -0,0 +1,376 @@
+                                         max_depth=self.max_depth,
+                                         min_samples_split=self.min_samples_split,
+                                         min_samples_leaf=self.min_samples_leaf,
+                                         min_weight_fraction_leaf=self.min_weight_fraction_leaf,
+                                         max_features=None,
+                                         random_state=self.random_state,
+                                         max_leaf_nodes=self.max_leaf_nodes,
+                                         min_impurity_decrease=self.min_impurity_decrease,
+                                         min_impurity_split=self.min_impurity_split)
+            tree.fit(features, lambdas)
+
+            self.ensemble.append(tree)
+
+            prediction = tree.predict(features)
+            model_preds += self.learning_rate * prediction
+            #TODO: Remove the next two statements after debugging
+            train_score = self._score(model_preds, scores, queries, 10)
+            print("  --iteration train score " + str(train_score))
+        return self.ensemble
+
+    def _predict_scores_fixed(self, X, **kwargs):
+        """
+            Predict the scores for a given collection of sets of objects of same size.
+
+            Parameters
+            ----------
+            X : array-like, shape (n_samples, n_objects, n_features)
+
+
+            Returns
+            -------
+            Y : array-like, shape (n_samples, n_objects)
+                Returns the scores of each of the objects for each of the samples.
+        """
+        n_instances, n_objects, n_features = X.shape
+        self.logger.info("For Test instances {} objects {} features {}".format(*X.shape))
+        X1 = X.reshape(n_instances * n_objects, n_features)
+        scores = np.zeros(n_instances * n_objects)
+        for tree in self.ensemble:
+            scores += tree.predict(X1)
+        scores = scores.reshape(n_instances, n_objects)
+        return scores
+
+    def predict_scores(self, X, **kwargs):
+        """
+            Predict the utility scores for each object in the collection of set of objects called a query set.
+
+            Parameters
+            ----------
+            X : numpy array of size (n_instances, n_objects, n_features)
+
+            Returns
+            -------
+            Numpy array of size (n_instances, n_objects)
+        """
+        return super().predict_scores(X, **kwargs)
+
+    def predict_for_scores(self, scores, **kwargs):
+        """
+         Predict rankings for the scores for a given collection of sets of objects (query sets). Wrapper that calls the function of the same name 
+         belonging to the ObjectRanker super class.
+        """
+        return ObjectRanker.predict_for_scores(self, scores, **kwargs)
+
+    def predict(self, X, **kwargs):
+        return super().predict(X, **kwargs)
```

Comment:
Running the predict method with a simple two feature problem does not work:
```python
from csrank import ObjectRankingDatasetGenerator
seed = 123
n_train = 10000
n_test = 10000
n_features = 2
n_objects = 5
gen = ObjectRankingDatasetGenerator(dataset_type='medoid', random_state=seed,
                                    n_train_instances=n_train,
                                    n_test_instances=n_test,
                                    n_objects=n_objects,
                                    n_features=n_features)
```

```
~/anaconda3/envs/csrank/lib/python3.6/site-packages/sklearn/tree/tree.py in _validate_X_predict(self, X, check_input)
    400                              "match the input. Model n_features is %s and "
    401                              "input n_features is %s "
--> 402                              % (self.n_features_, n_features))
    403 
    404         return X

ValueError: Number of features of the model must match the input. Model n_features is 1 and input n_features is 2
```

Please fix the bug.